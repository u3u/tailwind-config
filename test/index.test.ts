import { expect, it } from 'vitest';

it('should match snapshot', async () => {
  const { execa } = await import('execa');

  const { stdout } = await execa('tailwindcss', ['-i', 'test/__fixtures__/globals.css']);

  expect(stdout).toMatchSnapshot();
});
