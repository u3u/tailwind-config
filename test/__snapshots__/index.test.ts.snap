// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
"/* @tailwind base;
@tailwind components;
@tailwind utilities; */

[type='file'] {
  display: none;
}

[type='search'] {
  -webkit-appearance: none;
     -moz-appearance: none;
          appearance: none;
}

a {
  cursor: pointer;
}

a:hover {
  --tw-text-opacity: 1;
  color: rgb(26 102 254 / var(--tw-text-opacity));
}

a:active {
  --tw-text-opacity: 1 !important;
  color: rgb(11 73 217 / var(--tw-text-opacity)) !important;
}

a[type='button'] {
  -webkit-appearance: none;
     -moz-appearance: none;
          appearance: none;
}

a,
button {
  transition-property: color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, -webkit-backdrop-filter;
  transition-property: color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;
  transition-property: color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter, -webkit-backdrop-filter;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-duration: 300ms;
}

body {
  font-feature-settings: normal;
}

html {
  scroll-behavior: smooth;
  color: rgb(0 0 0 / 0.8);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

html,
body,
#__next {
  height: 100%;
}

input::-moz-placeholder {
  --tw-text-opacity: 1;
  color: rgb(150 150 150 / var(--tw-text-opacity));
}

input::placeholder {
  --tw-text-opacity: 1;
  color: rgb(150 150 150 / var(--tw-text-opacity));
}

input:-internal-autofill-selected {
  --tw-shadow: 0 0 0px 50px #fff inset;
  --tw-shadow-colored: inset 0 0 0px 50px var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
}

.btn {
  display: inline-flex;
  height: 3rem;
  min-height: 3rem;
  flex-shrink: 0;
  cursor: pointer;
  -webkit-user-select: none;
     -moz-user-select: none;
          user-select: none;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  border-radius: var(--rounded-btn, 0.5rem);
  border-color: transparent;
  border-color: oklch(var(--btn-color, var(--b2)) / var(--tw-border-opacity, 1));
  padding-left: 1rem;
  padding-right: 1rem;
  text-align: center;
  font-size: 0.875rem;
  line-height: 1em;
  gap: 0.5rem;
  font-weight: 600;
  text-decoration-line: none;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
  border-width: var(--border-btn, 1px);
  animation: button-pop var(--animation-btn, 0.25s) ease-out;
  transition-property: color, background-color, border-color, opacity, box-shadow, transform;
  --tw-text-opacity: 1;
  color: var(--fallback-bc,oklch(var(--bc)/var(--tw-text-opacity)));
  --tw-shadow: 0 1px 2px 0 rgb(0 0 0 / 0.05);
  --tw-shadow-colored: 0 1px 2px 0 var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
  outline-color: var(--fallback-bc,oklch(var(--bc)/1));
  background-color: oklch(var(--btn-color, var(--b2)) / var(--tw-bg-opacity, 1));
}

.btn[disabled],.btn:disabled {
  pointer-events: none;
}

:where(.btn:is(input[type=\\"checkbox\\"])),
:where(.btn:is(input[type=\\"radio\\"])) {
  width: auto;
  -webkit-appearance: none;
     -moz-appearance: none;
          appearance: none;
}

.btn:is(input[type=\\"checkbox\\"]):after,.btn:is(input[type=\\"radio\\"]):after {
  --tw-content: attr(aria-label);
  content: var(--tw-content);
}

@media (hover: hover) {
  .btn:hover {
    --tw-border-opacity: 1;
    border-color: var(--fallback-b3,oklch(var(--b3)/var(--tw-border-opacity)));
    border-color: color-mix(
          in oklab,
          oklch(var(--btn-color, var(--b2)) / var(--tw-border-opacity, 1)) 90%,
          black
        );
    --tw-bg-opacity: 1;
    background-color: var(--fallback-b3,oklch(var(--b3)/var(--tw-bg-opacity)));
    background-color: color-mix(
          in oklab,
          oklch(var(--btn-color, var(--b2)) / var(--tw-bg-opacity, 1)) 90%,
          black
        );
  }

  @supports not (color: oklch(0 0 0)) {
    .btn:hover {
      background-color: var(--btn-color, var(--fallback-b2));
      border-color: var(--btn-color, var(--fallback-b2));
    }
  }

  .btn:hover {
    --tw-border-opacity: 1;
    border-color: var(--fallback-b3,oklch(var(--b3)/var(--tw-border-opacity)));
    border-color: color-mix(
          in oklab,
          oklch(var(--btn-color, var(--b2)) / var(--tw-border-opacity, 1)) 90%,
          black
        );
    --tw-bg-opacity: 1;
    background-color: var(--fallback-b3,oklch(var(--b3)/var(--tw-bg-opacity)));
    background-color: color-mix(
          in oklab,
          oklch(var(--btn-color, var(--b2)) / var(--tw-bg-opacity, 1)) 90%,
          black
        );
  }

  @supports not (color: oklch(0 0 0)) {
    .btn:hover {
      background-color: var(--btn-color, var(--fallback-b2));
      border-color: var(--btn-color, var(--fallback-b2));
    }
  }

  .btn.daisy-glass:hover {
    --glass-opacity: 25%;
    --glass-border-opacity: 15%;
  }

  .daisy-btn-outline.btn:hover {
    --tw-text-opacity: 1;
    color: var(--fallback-pc,oklch(var(--pc)/var(--tw-text-opacity)));
    background-color: color-mix(in oklab, var(--fallback-p,oklch(var(--p)/1)) 90%, black);
    border-color: color-mix(in oklab, var(--fallback-p,oklch(var(--p)/1)) 90%, black);
  }

  .btn[disabled]:hover,.btn:disabled:hover {
    --tw-border-opacity: 0;
    background-color: var(--fallback-n,oklch(var(--n)/var(--tw-bg-opacity)));
    --tw-bg-opacity: 0.2;
    color: var(--fallback-bc,oklch(var(--bc)/var(--tw-text-opacity)));
    --tw-text-opacity: 0.2;
  }

  .btn:is(input[type=\\"checkbox\\"]:checked):hover,.btn:is(input[type=\\"radio\\"]:checked):hover {
    background-color: color-mix(in oklab, var(--fallback-p,oklch(var(--p)/1)) 90%, black);
    border-color: color-mix(in oklab, var(--fallback-p,oklch(var(--p)/1)) 90%, black);
  }
}

.btn:active:hover,.btn:active:focus {
  animation: button-pop 0s ease-out;
  transform: scale(var(--btn-focus-scale, 0.97));
}

@supports not (color: oklch(0 0 0)) {
  .btn {
    background-color: var(--btn-color, var(--fallback-b2));
    border-color: var(--btn-color, var(--fallback-b2));
    --btn-color: var(--fallback-p);
  }
}

.btn:focus-visible {
  outline-style: solid;
  outline-width: 2px;
  outline-offset: 2px;
}

.btn {
  --tw-text-opacity: 1;
  color: var(--fallback-pc,oklch(var(--pc)/var(--tw-text-opacity)));
  outline-color: var(--fallback-p,oklch(var(--p)/1));
}

@supports (color: oklch(0 0 0)) {
  .btn {
    --btn-color: var(--p);
  }
}

.btn.daisy-glass {
  --tw-shadow: 0 0 #0000;
  --tw-shadow-colored: 0 0 #0000;
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
  outline-color: currentColor;
}

.btn.daisy-glass.daisy-btn-active {
  --glass-opacity: 25%;
  --glass-border-opacity: 15%;
}

.daisy-btn-outline.btn {
  --tw-text-opacity: 1;
  color: var(--fallback-p,oklch(var(--p)/var(--tw-text-opacity)));
}

.daisy-btn-outline.btn.daisy-btn-active {
  --tw-text-opacity: 1;
  color: var(--fallback-pc,oklch(var(--pc)/var(--tw-text-opacity)));
  background-color: color-mix(in oklab, var(--fallback-p,oklch(var(--p)/1)) 90%, black);
  border-color: color-mix(in oklab, var(--fallback-p,oklch(var(--p)/1)) 90%, black);
}

.btn.daisy-btn-disabled,.btn[disabled],.btn:disabled {
  --tw-border-opacity: 0;
  background-color: var(--fallback-n,oklch(var(--n)/var(--tw-bg-opacity)));
  --tw-bg-opacity: 0.2;
  color: var(--fallback-bc,oklch(var(--bc)/var(--tw-text-opacity)));
  --tw-text-opacity: 0.2;
}

.btn:is(input[type=\\"checkbox\\"]:checked),.btn:is(input[type=\\"radio\\"]:checked) {
  --tw-border-opacity: 1;
  border-color: var(--fallback-p,oklch(var(--p)/var(--tw-border-opacity)));
  --tw-bg-opacity: 1;
  background-color: var(--fallback-p,oklch(var(--p)/var(--tw-bg-opacity)));
  --tw-text-opacity: 1;
  color: var(--fallback-pc,oklch(var(--pc)/var(--tw-text-opacity)));
}

.btn:is(input[type=\\"checkbox\\"]:checked):focus-visible,.btn:is(input[type=\\"radio\\"]:checked):focus-visible {
  outline-color: var(--fallback-p,oklch(var(--p)/1));
}"
`;
